### PR TITLE
add sqlite3_database class and it's implementation; add Makefile

### PR DESCRIPTION
**Pull request by RifeX-LA**

### CODE REVIEW

File: include/sqlite3.h [hunk 1]
```diff
@@ -0,0 +1,12806 @@
...
+** ^If the most recent evaluation of the statement encountered no errors
+** or if the statement is never been evaluated, then sqlite3_finalize() returns
+** SQLITE_OK.  ^If the most recent evaluation of statement S failed, then
+** sqlite3_finalize(S) returns the appropriate [error code] or
+** [extended error code].
+**
+** ^The sqlite3_finalize(S) routine can be called at any point during
+** the life cycle of [prepared statement] S:
+** before statement S is ever evaluated, after
+** one or more calls to [sqlite3_reset()], or after a
```
Comment (Temnyakov) [inline include/sqlite3.h:6109]: Sleep is good

- [x] Sleep
@woodenpencil

**Review by Temnyakov — COMMENTED** [review]
Good, verygood!
Don't forget to sleep!

**Review by Temnyakov — COMMENTED** [review]
Let's merge this

**Review by Temnyakov — APPROVED** [review]

**Review by MaximOn123 — APPROVED** [review]
Great work, colleague!

**Review by woodenpencil — COMMENTED** [review]
cool

**Pull request merged**